File: day04/star07.py
def check_line(line):
    line = line.strip()
    xs, ys = line.split(',')
    x1, x2 = map(int, xs.split('-'))
    y1, y2 = map(int, ys.split('-'))

    return (x1 <= y1 and y2 <= x2) or (y1 <= x1 and x2 <= y2)

n = 0
with open('input.txt') as f:
    for line in f:
        if check_line(line):
            n += 1

print(f"{n = }")

File: day04/star07.c
#include <stdio.h>
#include <stdlib.h>

int check_line(int x1, int x2, int y1, int y2) {
    return (x1 <= y1 && y2 <= x2) || (y1 <= x1 && x2 <= y2);
}

int main(void) {
    FILE *fp = fopen("input.txt", "r");
    int x1, x2, y1, y2;
    int n = 0;

    while (fscanf(fp, "%d-%d,%d-%d\n", &x1, &x2, &y1, &y2) == 4) {
        if (check_line(x1, x2, y1, y2)) n++;
    }

    printf("n = %d\n", n);
}